src/main/java/edu/harvard/iq/dataverse/export/openaire/OpenAireExportUtil.java

```diff
@@ -1355,22 +1359,36 @@ public static void writeGeoLocationsElement(XMLStreamWriter xmlw, DatasetVersion
     /**
      * 18 GeoLocation (R)
      *
+     * Write geoLocationPlace inside geoLocation element
+     * 
      * @param xmlw The Steam writer
-     * @param foo
-     * @param geoLocationPlace
+     * @param geoLocationPlace Geo location place
      * @param language current language
      * @throws XMLStreamException
      */
-    public static void writeGeoLocationsElement(XMLStreamWriter xmlw, Set<FieldDTO> foo, String geoLocationPlace, String language) throws XMLStreamException {
-        //boolean geoLocations_check = false;
+    public static void writeGeolocationPlace(XMLStreamWriter xmlw, String geoLocationPlace, String language) throws XMLStreamException {
         boolean geoLocation_check = false;
-        boolean geoLocationbox_check = false;
-
+        
         if (StringUtils.isNotBlank(geoLocationPlace)) {
-
             geoLocation_check = writeOpenTag(xmlw, "geoLocation", geoLocation_check);
             writeFullElement(xmlw, null, "geoLocationPlace", null, geoLocationPlace, language);
         }
+        writeEndTag(xmlw, geoLocation_check);
+    }
+    
+    /**
+     * 18 GeoLocation (R)
+     *
+     * @param xmlw The Steam writer
+     * @param foo
+     * @param language current language
+     * @throws XMLStreamException
+     */
+    public static void writeGeoLocationsElement(XMLStreamWriter xmlw, Set<FieldDTO> foo, String language) throws XMLStreamException {
```

Comment:
Could "set" be used instead of "foo"?

The pattern used in the code is the following:
```
for (HashSet<FieldDTO> foo : fieldDTO.getMultipleCompound()) {
         String awardNumber = null;
         String funderName = null;

          for (Iterator<FieldDTO> iterator = foo.iterator(); iterator.hasNext();) {
                 FieldDTO next = iterator.next();
```
Here the variable "foo" is used as a "general set".

Reply:
It can be done (about 35 matches on OpenAireExportUtil or 17 code refactor).

Reply:
I've committed the new version with the required changes.